### ISSUE TITLE
Request form: Focus reset on 'Add another site' button

### ISSUE THREAD

**Issue by AnnaGingle** [cisagov/getgov#2822]
### Current Behavior

On the Current websites page of the request form pressing the `Add another site` button resets the focus order to the breadcrumb. This fails Trusted Tester success criteria [2.4.3-focus-order-reveal](https://section508coordinators.github.io/TrustedTester/keyboard.html) and WCAG success criteria (SC) [2.4.3 Focus Order (Level A)](https://www.w3.org/WAI/WCAG21/Understanding/focus-order.html)

https://github.com/user-attachments/assets/d62c80ac-d2a8-4767-96b9-1ed3817874dd

``

### Expected Behavior

Focus should remain on the `Add another website` button. See the .gov domain page's Alternative domains (optional) section for an example of expected behavior.

https://github.com/user-attachments/assets/fb25779e-9dd5-4a99-b310-67af7741e9fc

### Steps to Reproduce

1. Start a domain request
2. On the Current websites page enter a website for What are the current websites for your organization?
3. Navigate with your keyboard to the `Add another site` button
4. Press enter on the `Add another site` button

### Environment

Environment: Staging
Browser: Chrome

### Prioritization Comments
**WCAG Conformance Level:** 2.1 AA
**WCAG Standard that applies:** 
2.4.3 - Focus Order (Critical)
**Critical Failure if Not Met?:**  In this case no, see below
**Prioritization:** High/Must Do
**Reason:** The current behavior where focus resets to the breadcrumb after pressing the "Add another site" button disrupts the user's navigation flow, especially for keyboard-only users and those relying on assistive technologies. This issue makes completing the form more cumbersome, causing potential confusion and frustration. While it does not block form completion, it significantly affects usability, making it a priority to resolve.
